dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -340,18 +336,21 @@ def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) ->
         )
 
     def _get_table_update_sql(
-        self, table_name: str, new_columns: Sequence[TColumnSchema], generate_alter: bool
+        self,
+        table_name: str,
+        new_columns: Sequence[TColumnSchema],
+        generate_alter: bool,
     ) -> List[str]:
         table: TTableSchema = self.prepare_load_table(table_name, self.in_staging_mode)
         sql = SqlJobClientBase._get_table_update_sql(self, table_name, new_columns, generate_alter)
 
         if generate_alter:
             return sql
 
-        # Default to 'ReplicatedMergeTree' if user didn't explicitly set a table engine hint.
-        table_type = cast(
-            TTableEngineType, table.get(TABLE_ENGINE_TYPE_HINT, "replicated_merge_tree")
-        )
+        # Default to 'MergeTree' if the user didn't explicitly set a table engine hint.
```

Comment:
doing this in adapter is really cool. just one thought? should we add this to clickhouse configuration anyway? so users can change the engine for all the resources without the need for adapter for each one? it will be similar to other `dlt` settings ie. where you can `iceberg` table format in athena for all resources (and also per resource with the decorator)

Reply:
@rudolfix Added!

https://github.com/dlt-hub/dlt/blob/f852d44d41ff77327a4735772ccd8f7affb03bb3/dlt/destinations/impl/clickhouse/configuration.py#L43-L44